src/components/SaveGameForm.js

```diff
@@ -87,22 +89,70 @@ export default function SaveGameForm({onSubmit}) {
     function handleChange(event) {
         setFormInput({
             ...formInput,
-            [event.target.name]: event.target.value,
+            [event.target.name]: event.target.value
         })
     }
 
     function handleSubmit(event) {
         event.preventDefault()
-        onSubmit(formInput)
+        const validation = validateForm(formInput)
+        if (validation.result) {
+            trimInputs(formInput)
+            onSubmit(formInput)
+            setFormInput({
+                location: '',
+                date: '',
+                players: '',
+                winner: '',
+                shots:'',
+            })
+            setErrorMessage('')
+        } else {
+           setErrorMessage(validation.message)
+        }
+    }
+
+    function validateForm(formInput) {
+       const filledInputs = validateNotEmpty(formInput)
+       const validNumber = validateShots(formInput)
+
+       if (filledInputs.result) {
+           if (validNumber.result) {
+                return {result: true}
+           } else {
+               return validNumber
+           }
+       } else {
+           return filledInputs
+       }
```

Comment:
Can't you make this a bit more compact with logical operators?

Reply:
One idea could be to rewrite it into two terneries. But I am not sure, if this is easier to follow, when two terneries are nested within each other? 
``const errorMessage = filledInputs.isValid ? (validNumber.isValid ? {isValid: true} : validNumber) : filledInputs``